docling_core/types/doc/document.py

```diff
@@ -2348,6 +2701,71 @@ def iterate_items(
                     included_content_layers=my_layers,
                 )
 
+    def iterate_items_with_stack(
+        self,
+        root: Optional[NodeItem] = None,
+        with_groups: bool = False,
+        traverse_pictures: bool = False,
+        page_no: Optional[int] = None,
+        included_content_layers: set[ContentLayer] = DEFAULT_CONTENT_LAYERS,
```

Comment:
Mutable default arguments is [bad practice](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments).

Reply:
fixed that